Examples/text-size-test.lisp

```diff
@@ -72,123 +144,118 @@
-	(draw-vstrecke stream
-		      (- x1 40)
-		      y1
-		      (+ y1 (text-style-height style medium))
-		      :line-style (make-line-style :thickness 2)
-		      :ink +black+)
-	(draw-hstrecke stream
-		       (- y1 20)
-		       x1
-		       (+ x1 (text-style-width style medium))
-		       :ink +black+)
-	(draw-line* stream
-		    0 ybase
-		    pane-width ybase
-		    :ink +green+)
-	(draw-text* stream str x1 ybase :text-style style)
-	;; Gtkairo's DRAW-TEXT* understands multiple lines.
-	;; (CLIM-CLX doesn't like multiple lines much.)
-	;;
-	;; If we use WRITE-STRING instead of DRAW-TEXT, the frontend will
-	;; handle the line breaks, but lines 2..n will start at x = 0 rather
-	;; than x = x1, confusing our diagram.
-;;;	(setf (stream-cursor-position stream) (values x1 y1))
-;;;	(with-text-style (stream style)
-;;;	  (write-string str stream))
-        (ecase rectangle
-          ((:text-size)
-           (draw-rectangle* stream
-                            x1 y1
-                            (+ x1 width) (+ y1 height)
-                            :ink +red+
-                            :filled nil)
-	    (draw-line* stream
-			0 (+ y1 final-y)
-			pane-width (+ y1 final-y)
-			:ink +blue+)
-	    (draw-line* stream
-			(+ x1 final-x) 0
-			(+ x1 final-x) pane-height
-			:ink +blue+))
-          ((:text-bounding-rectangle)
-           (multiple-value-bind (left top right bottom)
-               (climi::text-bounding-rectangle* medium str :text-style style)
-             (draw-rectangle* stream 
-                              (+ x1 left) (+ y1 baseline top)
-                              (+ x1 right) (+ y1 baseline bottom)
-                              :ink +purple+
-                              :filled nil))))))))
-
-(define-text-size-test-command (com-quit-text-size-test :menu "Quit") ()
-  (frame-exit *application-frame*))
-
-(define-text-size-test-command (com-update-text-size-test :menu "Update") ()
-  (display-canvas *application-frame*
-		  (frame-standard-output *application-frame*)))
+             (y1 (/ (- pane-height height) 2))
+             (ybase (+ y1 baseline)))
+
+        (draw-text* stream
+                    (format nil "fixed-width-p: ~(~A~)"
+                            (handler-case
+                                (text-style-fixed-width-p style medium)
+                              (error (c)
+                                c)))
+                    2
+                    pane-height
+                    :text-style (legend-text-style))
+        (labels ((component (thunk &rest args &key title &allow-other-keys)
+                   (let* ((ink             (prog1 (aref inks ink)
+                                             (incf ink)))
+                          (drawing-options (list* :ink ink (alexandria:remove-from-plist args :title))))
+                     (push (cons title drawing-options) legend)
+                     (apply #'invoke-with-drawing-options stream thunk drawing-options))))
```

Comment:
OK, so here's the problem (I think that's what you were mostly wondering about):

when you call it like this:

```
(with-drawing-options (stream …)
  (funcall thunk stream))
```
invoke-with-drawing-options is called with this body closure:
```
…
  (flet ((continuation (arg)
    (declare (ignore arg))
    (funcall thunk ***stream***))
…
```

OTOH, if you pass thunk (not a closure capturing stream) to invoke-with-drawing-options it goes like this:

```
invoke-with-drawing-options sheet continuation
-> [set the medium options]
--> (funcall thunk (sheet-medium sheet))
```

as seen here:
```
(defmethod invoke-with-drawing-options ((sheet sheet) continuation
					&rest drawing-options)
  (with-sheet-medium (medium sheet)
    (with-medium-options (medium drawing-options)
      (funcall continuation medium))))
```

I'm inclined to believe, that we should call continuation with the original sheet, although in the spec we have a passage: `continuation is a function of one argument, the medium;`[1], so that requires some consideration. If we decide to call it with the original sheet (as passed to the method), appropriate annotation should be added to [1].

So having this in mind, the problem is that medium is passed to thunk, so no output is recorded on the stream. Repaint replays recorded output.

http://bauhh.dyndns.org:8000/clim-spec/10-2.html#_526

Reply:
As I said: thanks for investigating. I added the fixed.

However, for the text size text, I went with repainting instead of output recording in the end as this makes everything easier.